src/Orckestra.Composer.Website/App_Code/FacetHelpers.cshtml

```diff
@@ -119,55 +110,104 @@
     </button>
 }
 
-@helper ShowMoreLessFacets(Facet facet)
+@helper RenderFacet()
 {
-    <ul id="onDemandFacets-@facet.FieldName" class="list-unstyled  collapse">
-        @if (facet.FacetType == FacetType.MultiSelect)
-        {
-            foreach (var facetValue in facet.OnDemandFacetValues)
-            {
-                @MultiSelectFacet(facetValue, facet)
-            }
-        }
-
-        @if (facet.FacetType == FacetType.SingleSelect)
-        {
-            foreach (var facetValue in facet.OnDemandFacetValues)
-            {
-                @SingleFacet(facetValue, facet);
-            }
-        }
-    </ul>
-    @ShowMoreLessButton(facet.FieldName)
+    <div v-if="facet.FacetType == '@(FacetType.SingleSelect)'">
+        @SingleFacet()
+    </div>
+    <div v-if="facet.FacetType == '@(FacetType.MultiSelect)'">
+        @MultiSelectFacet()
+    </div>
+    <div v-if="facet.FacetType == '@(FacetType.Range)'">
+        @RangeFacet()
+    </div>
+    <div v-if="facet.FacetType != '@(FacetType.Range)'">
```

Comment:
Is it normal ShowMoreLessFacets shoed when SingleSelect or MultiSelect?

Reply:
yes, for Range Facet we do not show this